components/submissions/StatusBadge.tsx

```diff
@@ -0,0 +1,20 @@
+import { FC } from "react";
+
+interface Props {
+    status: string
+}
+
+export const StatusBadge:FC<Props> = ({ status }) => {
+
+    const statusMap = new Map([
```

Comment:
Ok so, couple of things:

1. Data that doesn't depend on react state or props should be defined outside the component. Right now you're declaring a new map on every render cycle of react for no good reason
2. The Map data type should be used when you don't know the shape of the data you need to map. In this case you definitely know the full shape of the object and should therefore be just a simple constant object